### PR TITLE
Ganfra/kotlin binding scripts

### PR DESCRIPTION
**Pull request by ganfra**
This PR adds a kotlin xtask to build the so files and generate the kotlin ffi bindings (instead of using a shell script).
It also removes all the kotlin specific code to be moved back to https://github.com/matrix-org/matrix-rust-components-kotlin

### CODE REVIEW

File: xtask/src/kotlin.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    Ok(())
+}
+
+fn build_for_android_target(
+    target: &str,
+    profile: &str,
+    dest_dir: &str,
+    package_name: &str,
+) -> Result<PathBuf> {
+    cmd!("cargo ndk --target {target} -o {dest_dir} build --profile {profile} -p {package_name}")
```
Comment (Hywan) [inline xtask/src/kotlin.rs:157]: We need to install [`cargo-ndk`](https://github.com/bbqsrc/cargo-ndk) on the CI, as long as all the targets/toolchains passed to this function.

**Review by Hywan — COMMENTED** [review]
It's shaping really great, thanks for simplifying all that!

File: xtask/src/kotlin.rs (hunk 1 above)
Comment (ganfra) [reply xtask/src/kotlin.rs:157]: I think all the targets are already installed, but yes it's then missing the cargo ndk

Comment (Hywan) [reply xtask/src/kotlin.rs:157]: Then no :-)

**Review by Hywan — APPROVED** [review]

**Pull request merged**